.github/workflows/windows.yml

```diff
@@ -1,77 +1,147 @@
 name: Windows builds
 on: push
-env:
-  MAX_WARNINGS_GCC_32bit_Debug:     229
-  MAX_WARNINGS_GCC_64bit_Debug:     304
-  MAX_WARNINGS_Clang_32bit_Debug:    98
-  MAX_WARNINGS_Clang_64bit_Debug:   142
 
 jobs:
 
-  build_msvc_matrix:
-    name: MSVC 32-bit (${{ matrix.type }})
+  build_windows_msys2:
+    name: ${{ matrix.conf.compiler }} ${{ matrix.conf.bits }}-bit
     runs-on: windows-latest
     strategy:
       matrix:
-        type: [Release, Debug]
-    steps:
-      - uses:  actions/checkout@v1
-      - name:  Install packages
-        shell: pwsh
-        run: |
-          vcpkg install libpng sdl1 sdl1-net opusfile
-          vcpkg integrate install
-      - name:  Log environment
-        shell: pwsh
-        run:   .\scripts\log-env.ps1
-      - name:  ${{ matrix.type }} build
-        shell: pwsh
-        env:
-          PATH: '${env:PATH};C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\amd64'
-        run: |
-          cd visualc_net
-          MSBuild -m dosbox.sln -p:Configuration=${{ matrix.type }}
-
-
-  build_msys2_matrix:
-    name: ${{ matrix.compiler }} ${{ matrix.bits }}-bit
-    runs-on: windows-latest
-    strategy:
-      matrix:
-        compiler: [GCC, Clang]
-        bits:     [32, 64]
+        conf:
+          - compiler: GCC
+            bits: 32
+            arch: i686
+            max_warnings: 229
+          - compiler: GCC
+            bits: 64
+            arch: x86_64
+            max_warnings: 304
+          - compiler: Clang
+            bits: 32
+            arch: i686
+            max_warnings: 98
+          - compiler: Clang
+            bits: 64
+            arch: x86_64
+            max_warnings: 142
     env:
       CHERE_INVOKING: yes
     steps:
       - uses:  actions/checkout@v1
       - uses:  actions/cache@v1
```

Comment:
With your variable export technique, we can now roll over our cache once a week :-)
New step before `actions/cache`:
``` yaml
- name: Get Date
  id: get-date
  run: echo "::set-output name=date::$(/bin/date -u "%Y-%W")"
  shell: bash
```